src/librustc_errors/emitter.rs

```diff
@@ -311,7 +311,11 @@ impl EmitterWriter {
         if line.annotations.len() == 1 {
             if let Some(ref ann) = line.annotations.get(0) {
                 if let AnnotationType::MultilineStart(depth) = ann.annotation_type {
-                    if source_string[0..ann.start_col].trim() == "" {
```

Comment:
Why is the byte offset unavailable here? Converting to column should be delayed until the last moment IMO (and should take into account unicode grapheme width).

Reply:
@eddyb I'm not sure what you mean but the start_col is given as columns, as per this comment: https://github.com/rust-lang/rust/blob/32b50e280faf56f21cbd82d1cf82cb4795535143/src/librustc_errors/snippet.rs#L123-L124